config/stack/manifests/resources/minimalgcp.ui-schema.yaml

```diff
@@ -1,57 +1,68 @@
   items:
-  - name: region
-    controlType: singleInput
-    type: string
-    path: .spec.region
-    title: Region
-    description: The region that the resources will be deployed to
-    default: us-central1
-    validation:
-    - minLength: 2
-    - maxLength: 62
   - name: projectID
     controlType: singleInput
     type: string
     path: .spec.projectID
     title: Project ID
-    pattern: ^([A-Za-z0-9](?:(?:[-A-Za-z0-9]){0,61}[A-Za-z0-9])?){2,62}$
-    description: The GCP project that the resources will be deployed in
-    validation:
-    - minLength: 2
-    - maxLength: 62
-  - name: credentialSecretName
-    controlType: singleInput
-    type: string
-    path: .spec.credentialsSecretRef.name
-    title: Credentials Secret Name
-    pattern: ^([A-Za-z0-9](?:(?:[-A-Za-z0-9]){0,61}[A-Za-z0-9])?){2,62}$
-    description: The name of the secret that you stored your GCP credentials.
+    pattern: ^[a-z][-A-Za-z0-9]{5,29}$
     validation:
-    - minLength: 2
-    - maxLength: 62
-  - name: credentialSecretNamespace
-    controlType: singleInput
-    type: string
-    path: .spec.credentialsSecretRef.namespace
-    title: Credentials Secret Namespace
-    pattern: ^([A-Za-z0-9](?:(?:[-A-Za-z0-9]){0,61}[A-Za-z0-9])?){2,62}$
-    description: The namespace of the secret that you stored your GCP credentials.
+    - required: true
+    - minLength: 6
+    - maxLength: 30
+  - name: region
+    controlType: singleSelect
+    path: .spec.region
+    title: Region
+    description: Region of the resources that will be deployed.
+    enum:
+    - 'asia-east1'
+    - 'asia-east2'
+    - 'asia-northeast1'
+    - 'asia-northeast2'
+    - 'asia-south1'
+    - 'asia-southeast1'
+    - 'australia-southeast1'
+    - 'europe-north1'
+    - 'europe-west1'
+    - 'europe-west2'
+    - 'europe-west3'
+    - 'europe-west4'
+    - 'europe-west6'
+    - 'northamerica-northeast1'
+    - 'southamerica-east1'
+    - 'us-central1'
+    - 'us-east1'
+    - 'us-east4'
+    - 'us-west1'
+    - 'us-west2'
     validation:
-    - minLength: 2
-    - maxLength: 62
-  - name: credentialSecretKey
-    controlType: singleInput
-    type: string
-    path: .spec.credentialsSecretRef.key
-    title: Credentials Secret Data Key
-    pattern: ^([A-Za-z0-9](?:(?:[-A-Za-z0-9]){0,61}[A-Za-z0-9])?){2,62}$
-    description: The key that has the credential data in the secret.
+    - required: true
+- title: Generate Provider Secret
+  description: A provider secret will need to be generated via the gcloud CLI tool using the following command.
+  items:
+  - name: additionalInfoSnippet
+    controlType: codeSnippet
+    codeText: curl -s https://raw.githubusercontent.com/crossplaneio/stack-gcp-minimal/master/hack/gcp-credentials.sh | PROJECT_ID=[PROJECT_ID] bash -
+- title: Enter Your Generated Provider Secret
+  description: Once you have successfully generated the provider secret using the command above you can now enter it below.
```

Comment:
Maybe this:
> Enter the GCP keyfile contents which can be generated using the command above.

This wording is biased by a thought that we may want to be more welcoming of keyfiles that the user has created on their own.  To support that we would want to link them to the docs that describe all the features we need in a keyfile bound to a specific SA (https://crossplane.io/docs/v0.3/cloud-providers/gcp/gcp-provider.html ).

Reply:
did you want to make this wording update @rathpc that @displague has suggested? sounds like good reasoning to me.

Reply:
synced up with @rathpc, we'll just keep the originally intended messaging/design for now.